lib/ansible/parsing/yaml/constructor.py

```diff
@@ -78,6 +78,14 @@ def construct_mapping(self, node, deep=False):
 
         return mapping
 
+    def construct_ansible_int(self, node, unsafe=False):
+        value = self.construct_scalar(node)
+        if value.startswith('0') and len(value) > 1 and not value.startswith('0x'):
```

Comment:
should this also have a conditional for `elif value.startswith('0x'):` ?

Reply:
i was considering it, but we don't seem to use hex numbers anywhere